core/lib/Cardano/Address/Script.hs

```diff
@@ -200,6 +228,62 @@ keyHashFromBytes bytes
     | BS.length bytes /= hashSize = Nothing
     | otherwise = Just $ KeyHash bytes
 
+data ErrorKeyHashFromText =
+      ErrorKeyHashFromTextInvalidString (AbstractEncoding ())
+    | ErrorKeyHashFromTextWrongEncoding
+    | ErrorKeyHashFromTextWrongPayload
+    | ErrorKeyHashFromTextWrongHrp
+    | ErrorKeyHashFromTextWrongDataPart
+    deriving (Show, Eq)
+
+showErr :: ErrorKeyHashFromText -> String
+showErr (ErrorKeyHashFromTextInvalidString EBase16) =
+    "Invalid Base16-encoded string."
+showErr (ErrorKeyHashFromTextInvalidString EBech32{}) =
+    "Invalid Bech32-encoded string."
+showErr (ErrorKeyHashFromTextInvalidString EBase58) =
+    "Invalid Base58-encoded string."
+showErr ErrorKeyHashFromTextWrongEncoding =
+    "Verification key hash must be must be encoded as \
+    \base16, bech32 or base58."
+showErr ErrorKeyHashFromTextWrongPayload =
+    "Verification key hash must contain exactly 28 bytes."
+showErr ErrorKeyHashFromTextWrongHrp =
+    "Verification key hash must have 'script_vkh' hrp when Bech32-encoded."
+showErr ErrorKeyHashFromTextWrongDataPart =
+    "Verification key hash is Bech32-encoded but has wrong data part."
+
+-- | Construct a 'KeyHash' from 'Text'. Either hex encoded text or
+-- Bech32 encoded text with `script_vkh` hrp is expected. Also
+-- binary payload is expected to be composed of 28 bytes.
+--
+-- @since 3.0.0
+keyHashFromText :: Text -> Either ErrorKeyHashFromText KeyHash
+keyHashFromText txt =  case detectEncoding str of
+        Just EBase16 -> case fromBase16 (toBytes str) of
+            Left _ -> Left $ ErrorKeyHashFromTextInvalidString EBase16
+            Right bytes -> checkPayload bytes
+        Just EBech32{} -> fromBech32
+        Just EBase58 -> case fromBase58 (toBytes str) of
+            Left _ -> Left $ ErrorKeyHashFromTextInvalidString EBase58
+            Right bytes -> checkPayload bytes
+        Nothing -> Left ErrorKeyHashFromTextWrongEncoding
+ where
+    str = T.unpack txt
+    toBytes = T.encodeUtf8 . T.pack
+    checkPayload bytes =
+        maybeToRight ErrorKeyHashFromTextWrongPayload (keyHashFromBytes bytes)
+    fromBech32 = do
+        (hrp, dp) <- either
+            (const $ Left $ ErrorKeyHashFromTextInvalidString $ EBech32 ())
+            Right (Bech32.decodeLenient txt)
+        case Bech32.humanReadablePartToText hrp of
+            "script_vkh" -> do
```

Comment:
Is this specified anywhere?

Reply:
yes, it it specified here -> https://github.com/cardano-foundation/CIPs/pull/31

Reply:
Thanks